src/microbe_stage/editor/TimelineTab.cs

```diff
@@ -0,0 +1,249 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Godot;
+
+public class TimelineTab : PanelContainer
+{
+    [Export]
+    public NodePath EventsContainerPath;
+
+    [Export]
+    public NodePath ScrollContainerPath;
+
+    [Export]
+    public NodePath LocalFilterButtonPath;
+
+    [Export]
+    public NodePath GlobalFilterButtonPath;
+
+    private MicrobeEditor editor;
+    private PatchMapDrawer patchMapDrawer;
+
+    private VBoxContainer eventsContainer;
+    private ScrollContainer scrollContainer;
+    private Button localFilterButton;
+    private Button globalFilterButton;
+
+    private Filters eventFilter = Filters.Local;
+
+    private List<TimelineSection> cachedLocalTimelineElements;
+    private List<TimelineSection> cachedGlobalTimelineElements;
+
+    public enum Filters
+    {
+        /// <summary>
+        ///   Means to show all events in the world.
+        /// </summary>
+        Global,
+
+        /// <summary>
+        ///   Means to only show events relevant to the patch the player is in.
+        /// </summary>
+        Local,
+    }
+
+    /// <summary>
+    ///   Selects what events should be shown in the timeline tab.
+    /// </summary>
+    public Filters EventFilter
+    {
+        get => eventFilter;
+        set
+        {
+            eventFilter = value;
+            ApplyEventsFilter();
+        }
+    }
+
+    public void Init(MicrobeEditor editor, PatchMapDrawer drawer)
```

Comment:
Passing this these objects is a bit more coupling than I'd like. Couldn't these be passed in the UpdateTimeline method?

Reply:
Done.